### ISSUE TITLE
 Standard names: *Norm of the horizontal gradient of the chlorophyll concentration*

### ISSUE THREAD

**Issue by abiardeau** [cf-convention/vocabularies#50]
Hello,
I am Aurore BIARDEAU from Mercator Ocean international, working for Copernicus Marine service.
Date : 13-05-2024

We are in the process of implementing a new variable in the Copernicus Marine service and we would be happy to have your feedback on our proposal :

**standard_name** : `mass_concentration_of_chlorophyll_a_in_sea_water_horizontal_gradient`

**unit** : mg / m3 / km

**definition** (from existing `mass_concentration_of_chlorophyll_a_in_sea_water`) : 
'Mass concentration' means mass per unit volume and is used in the construction mass_concentration_of_X_in_Y, where X is a material constituent of Y. A chemical or biological species denoted by X may be described by a single term such as 'nitrogen' or a phrase such as 'nox_expressed_as_nitrogen'. Chlorophylls are the green pigments found in most plants, algae and cyanobacteria; their presence is essential for photosynthesis to take place. There are several different forms of chlorophyll that occur naturally. All contain a chlorin ring (chemical formula C20H16N4) which gives the green pigment and a side chain whose structure varies. The naturally occurring forms of chlorophyll contain between 35 and 55 carbon atoms. Chlorophyll-a is the most commonly occurring form of natural chlorophyll. The chemical formula of chlorophyll-a is C55H72O5N4Mg.

I can provide more information from my expert colleagues if needed,
Kind regards,
Aurore

**Comment by abiardeau** [comment]
Hello both, 

I got some answers to Jonathan questions from the expert I am in contact with : 

- `I think that this order of phrases is hard to parse. Do you mean horizontal_gradient_of_mass_concentration_of_chlorophyll_a_in_sea_water?`


Yes I think we are talking about the scalar magnitude of the horizontal gradient

- `What does "horizontal gradient" mean? According to the [guidelines](https://cfconventions.org/Data/cf-standard-names/docs/guidelines.html), if is the magnitude of a vector gradient, it should be something like magnitude_of_horizontal_gradient. If it's a vector, we need components such as eastward_derivative and northward_derivative. Since your title says Norm of the horizontal gradient, I guess you mean the former. In either case, something should be said in the definition about this.`

According to the title "Norm of the horizontal gradient" we are not providing the vector quantity but rather a scalar field.

I will ask their opinion for your question about the units as well. 

Thanks very much, 
Aurore

**Comment by JonathanGregory** [comment]
Dear Aurore

By "norm", then, do you mean the `magnitude_of_horizontal_gradient` i.e. |**&nabla;** _X_|,  where _X_ is scalar?

Best wishes

Jonathan

**Comment by abiardeau** [comment]
Dear all, 

I just got feedback from the expert I am in contact with about this standard_name.
He confirms that we are talking about magnitude of horizontal gradient, and proposes : 

`magnitude_of_horizontal_gradient_of_mass_concentration_of_chlorophyll_a_in_sea_water`

with this explanation concerning the units : 

Regarding the units, we express them that way based on the units of CHL-a, which we provide in mg/m3. The distance dimension (as they are gradients in space) is expressed in km as that is the scale at which the variable has meaning, it is the order of magnitude of the pixel size.

For the exact implementation of the units field, we have proposed “milligram m-3 km-1” following the field provided for the CHL-a variables, it also works for the udunits2 package.

What do you think ? 
Aurore

**Comment by JonathanGregory** [comment]
Dear Aurore @abiardeau 

I think that standard name is fine, and correct for what you want. Its canonical unit should be `kg m-4`, since we always use SI base units for the canonical units, but it's fine to put `units="milligram m-3 km-1"` in the netCDF file, since they are dimensionally equivalent.

```
You have: milligram m-3 km-1
You want: kg m-4
    1 milligram m-3 km-1 = 1e-09 (kg m-4)
    x/(kg m-4) = 1e-09*(x/(milligram m-3 km-1))
```

Best wishes

Jonathan

**Comment by abiardeau** [comment]
Thanks Jonathan ! 
We will proceed with this.
Kind regards, 
Aurore

**Comment by efisher008** [comment]
Dear Aurore @abiardeau 

It looks as though this name has reached consensus and the units for the term in the editor have been changed to `kg m-4`. Once the period for further comments/issues has passed, this will be accepted and published in the next standard names update (scheduled for mid-November). 

The only question I have is for @JonathanGregory - there is no P06 term for the canonical unit `kg m-4`. How would you suggest to retrieve the NVS ID from the P06 vocab for this quantity?

Best wishes,
Ellie

**Comment by JonathanGregory** [comment]
Dear Ellie

> There is no P06 term for the canonical unit `kg m-4`. How would you suggest to retrieve the NVS ID from the P06 vocab for this quantity?

I'm sorry, I don't know. I don't think that CF should be constrained by the capabilities of the NERC vocabulary server, though. Such questions are being discussed in https://github.com/orgs/cf-convention/discussions/366.

Best wishes

Jonathan

**Comment by feggleton** [comment]
@efisher008 if the unit isn't in the P06 vocabulary you can request for it to be added by Gwen. If it is valid and should be in the units vocab.

**Comment by efisher008** [comment]
Thank you Fran @feggleton. I'll get in touch with Gwen about the new P06 unit.

**Comment by feggleton** [comment]
I disagree with the recommendation for units of kg m-4 as I think this doesn't mean anything to the user, I think it causes confusion about the parameter and we have examples in the standard name table of not applying SI units and not collapsing units when it makes the term clearer in how its calculated. I would suggest 'kg m-3 m-1'.

For example, this is similar to the dimensionless salinity terms when we use g kg-1 as it provides information about how the quantity itself was calculated. For example, sea_water_absolute_salinity.

Neither of the suggested units are on P06 so we will need to make a request either way. Then we can get this term accepted once that's agreed.

**Comment by abiardeau** [comment]
Thanks for your feedback on the unit, it seems relevant to me.

**Comment by larsbarring** [comment]
I support Fran's @feggleton suggestion to use   **kg m-3 m-1** as canonical units.

**Comment by efisher008** [comment]
Hi all,

It appears we have come to a consensus about the canonical units for this name. I think the next step would be for @feggleton or one of the standard name moderators to reach out to the NERC Vocabulary Server to make a request for the unit in question `kg m-3 m-1` to be added. This can then be attached to the standard name `magnitude_of_horizontal_gradient_of_mass_concentration_of_chlorophyll_a_in_sea_water.` As a reminder, the name and its proposed description can be viewed here: https://cfeditor.ceda.ac.uk/proposal/5339.

Best, 
Ellie

**Comment by taylor13** [comment]
Browsing the standard_name table, it appears that the "canonical" units invariable use "kg" rather than "g", for example for a measure of mass, and "m" rather than "km" for a measure of length.  that would argue for `kg m-3 m-1`, rather than, say, `mg m-3 km-1`.  The user can, of course, report the variable using whatever equivalent units are preferred.

P.S. I'm still curious why one would choose to report a gradient of a field as opposed to the field itself (along with the coordinate information) from which gradients could be calculated.  By reporting gradients, you clearly lose some of the original information about the field.

**Comment by larsbarring** [comment]
Regarding the unit I fully agree with Karl. 

Some quantities may be reported in mg m-3 or in ug dm-3, or in g m-3 or ug ml-3, or ... depending on community, measurement system, or which quantity it is. To avoid all this CF uses the concept of canonical units, which as Karl writes gives the data producer the freedom to choose any of the equivalent units, and similarly the data users the freedom to convert to whichever equivalent unit they feel is suitable. And the canonical units are expressed in terms of SI base units, SI derived units, or non-SIS units accepted for use with the SI units (from the latter group I think only bel (as in decibel) is used). There are exceptions where this is not possible or feasible, but it holds as a general rule-of-thumb.

Hence, I think that in this case the appropriate canonical unit is kg m-3 m-1.

**Comment by efisher008** [comment]
Hi all,

Thank you very much for the careful discussion and the effort in moving this forward. I have added the P06 vocabulary reference to the entry in the CF editor for the units `kg m-3 m-1`, and this name has been accepted. It will be published in the next version of the CF Standard Names table, which is scheduled for release in July 2025. 

Best wishes,
Ellie

**Comment by efisher008** [comment]
This name has been published in v92 of the Standard Name Table, dated 24 July 2025. I am now closing this issue as resolved.
